features/frameworks/greentea-client/source/test_env.cpp

```diff
@@ -61,8 +61,10 @@ static void greentea_notify_version();
  * Rawserial object used to provide direct, raw serial communications
  * between the target and the host.
  */
-RawSerial greentea_serial(USBTX, USBRX); 
-
+RawSerial& greentea_serial() { 
+    static RawSerial serial(USBTX, USBRX);
```

Comment:
My change was actually just merged in, but yeah this may be another alternative. I suppose one benefit of this is that the `RawSerial` instance is on the stack, not the heap. But yeah I think it'd be worth verifying your concerns you voiced above @c1728p9.